A79-non-per-call-metrics-architecture.md

```diff
@@ -0,0 +1,426 @@
+// instruments. This API is thread-unsafe.
+class GlobalInstrumentsRegistry {
+ public:
+  enum class ValueType {
+    kUndefined,
+    kUInt64,
+    kDouble,
+  };
+  enum class InstrumentType {
+    kUndefined,
+    kCounter,
+    kHistogram,
+  };
+  struct GlobalInstrumentDescriptor {
+    ValueType value_type;
+    InstrumentType instrument_type;
+    uint32_t index;
+    bool enable_by_default;
+    absl::string_view name;
+    absl::string_view description;
+    absl::string_view unit;
+    std::vector<absl::string_view> label_keys;
+    std::vector<absl::string_view> optional_label_keys;
+  };
+  struct GlobalInstrumentHandle {
+    uint32_t index;
+  };
+  struct GlobalUInt64CounterHandle : public GlobalInstrumentHandle {};
+  struct GlobalDoubleHistogramHandle : public GlobalInstrumentHandle {};
+
+  // Creates instrument in the GlobalInstrumentsRegistry.
+  static GlobalUInt64CounterHandle RegisterUInt64Counter(
+      absl::string_view name, absl::string_view description,
+      absl::string_view unit, absl::Span<const absl::string_view> label_keys,
+      absl::Span<const absl::string_view> optional_label_keys,
+      bool enable_by_default);
+  static GlobalDoubleHistogramHandle RegisterDoubleHistogram(
+      absl::string_view name, absl::string_view description,
+      absl::string_view unit, absl::Span<const absl::string_view> label_keys,
+      absl::Span<const absl::string_view> optional_label_keys,
+      bool enable_by_default);
+  static void ForEach(
+      absl::FunctionRef<void(const GlobalInstrumentDescriptor&)> f);
+};
+```
+
+##### Java
+
+(To be filled.)
+
+##### Go
+
+(To be filled.)
+
+#### Global Stats Plugin Registry
+
+A global stats plugin registry is added. This registry keeps track of all stats
+plugins that have been registered globally.
+
+The application will use a stats plugin builder (e.g.
+`OpenTelemetryPluginBuilder`) to register a stats plugin to the registry. On
+creation, stats plugins should query the global instruments registry to get the
+list of instrument descriptors. Based on these descriptors and the stats
+plugin's configuration, backing instruments (e.g., `OpenTelemetry Instruments`)
+will be created. Implementations can choose an alternative form of
+representation for the global stats plugin registry. For example, gRPC Go might
+choose to register global stats plugin through global dial options.
+
+![](A79_graphics/global-stats-plugin-registry-usage.png)
+
+##### Scoping Stats Plugins for Channels
+
+To allow stats plugins to scope themselves to a certain set of channels,
+implementations can choose one of two approaches.
+
+In the first approach, stats plugins provide a scope filter that applications
+can configure. Based on the channel's configuration, a stats plugin can decide
+whether it wants to record metrics for the channel. At channel creation time,
+the channel's target and the authority is provided to the stats plugin to make
+this decision.
```

Comment:
The following diagram has Stats Plugin group and the description here does not refer to the concept. Can you add it in the descroption?